test/system/autoload_test.rb

```diff
@@ -45,4 +55,14 @@ class AutoloadTest < ApplicationSystemTestCase
       assert_text "Namespace: Hello, from Turbo page"
     end
   end
+
+  def click_button(*)
```

Comment:
In its current state, this draft PR only extends the `click_button` method, but if we find this approach viable, we'd want to extend all methods in [Capybara::DSL](https://github.com/teamcapybara/capybara/blob/f7ab0b5cd5da86185816c2d5c30d58145fe654ed/lib/capybara/dsl.rb#L49-L62).

Reply:
This is a super interesting way to address this problem!  One thing that I'm wrestling with is that I can't help but feel like this behavior codifies the possible experience for a user where:

1. they click on a button, it doesn't work (controller hasn't loaded yet)
2. they wait for a second, click on it again, and it does work (after the controller has loaded)

Granted, in _most_ cases, the delay here should be negligible and shouldn't be a problem, but I do think there are situations where this will come up outside of an automated test scenario.

Just an idea here, but I wonder if it'd be possible to specify strategies for how to deal with this in the library that would help avoid the need to override this behavior in capybara, e.g. `data-controller-autoload="hide"` or `data-controller-autoload="spinner"` - in the former case, you could just wait for the element to become visible to click on it (which capybara already does pretty well), and in the latter, you could handle that with a helper method somewhere.  By making it an explicit attribute, I think it could help illustrate to devs that this node may render before the javascript behavior has been attached to it, and you may need to deal with that.